### PR TITLE
feat: [DHIS2-9666] Create new TEI when none of the results match 

### PR DESCRIPTION
**Pull request by jasminenguyennn**

### CODE REVIEW

File: src/core_modules/capture-core/components/DataEntries/TeiRegistrationEntry/hooks/useFormValuesFromSearchTerms.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// @flow
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { convertClientToForm } from '../../../../converters';
+
+
+export const useFormValuesFromSearchTerms = () => {
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/DataEntries/TeiRegistrationEntry/hooks/useFormValuesFromSearchTerms.js:7]: Hey @jasminenguyennn , 

I noticed that for TEAs where the search terms use the `From-To` format, the `[object Object]` value is displayed on the new TEI page. Some examples of these TEAs are the ones with the type `Date` or `Number`.

Maybe these fields should be left empty or pre-filled with either the `from` or `to` search terms?  WDYT? You can find more details in the below video. 


https://user-images.githubusercontent.com/6817073/194064429-665747ca-3647-43a3-aa28-3e0649d00c40.mov


Thank you!

File: src/core_modules/capture-core/converters/clientToForm.js [hunk 2]
```diff
@@ -70,6 +83,8 @@ const valueConvertersForType = {
     [dataElementTypes.TRUE_ONLY]: () => 'true',
     [dataElementTypes.BOOLEAN]: (rawValue: boolean) => (rawValue ? 'true' : 'false'),
     [dataElementTypes.AGE]: convertAgeForEdit,
+    [dataElementTypes.NUMBER_RANGE]: convertNumberRangeForDisplay,
+    [dataElementTypes.DATE_RANGE]: convertDateRangeForDisplay,
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/converters/clientToForm.js:87]: Hey, 

There are a couple of  `*_RANGE` TEAs missing like: 
``` 
    INTEGER_RANGE 
    INTEGER_POSITIVE_RANGE, 
    INTEGER_NEGATIVE_RANGE, 
    INTEGER_ZERO_OR_POSITIVE_RANGE, 
    TIME_RANGE 
``` 
The full list supported by the search component can be found in the [searchFormElementConverter.js](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchFormElementConverter/SearchFormElementConverter.js#L51). Does it make sense to include them here?  

Thanks!

Comment (jasminenguyennn) [reply src/core_modules/capture-core/converters/clientToForm.js:87]: hey, it seems like I can not find the way to add these *_RANGE TEAs via maintenance  app. Any chance that they are no longer supported? 

Screenshot 2022-10-07 at 08 28 18

Comment (jasminenguyennn) [reply src/core_modules/capture-core/converters/clientToForm.js:92]: we discussed this offline and I updated the converter to reflex the changes.

**Review by simonadomnisoru — APPROVED** [review]
LGTM!

**Comment by karolinelien** [comment]
Works as expected from a user perspective.

**Pull request merged**